### ISSUE TITLE
"The name does not exist in the current context" for standard tables

### ISSUE THREAD

**Issue by sgiordy** [Microsoft/AL#1581]
If you try to extend a page that refers to a standard table, you cannot access the record variable.
Example:

    pageextension 51000 P51000 extends "User Card"
    {
      trigger OnAfterGetRecord()
      begin
        message("User Security ID");
      end;
    }

During the Publish-NAVApp you get this error:

    error AL0118: The name 'User Security ID' does not exist in the current context

**Comment by dennisfrostlander** [comment]
This should definitely work. If everything compiles fine in VSCode, this could be a problem with application symbols on the server (e.g. the server compiler has no knowledge of fields in standard tables).
- What version of AL extension and Nav server do you have?
- What symbols do you have published on the server:
`Get-NAVAppInfo -SymbolsOnly`
- Are you using automatic symbols generation in C/SIDE (and EnableSymbolLoadingAtServerStartup setting on the server)?

**Comment by sgiordy** [comment]
We are working with NAV 2018 CU1 in the italian localization.

The AL extension is that provided on NAV DVD: 0.12.15898

The symbols has been already loaded:

    Id            : d4a14f4a-0ae8-4501-a527-b3e36c0c3d67
    Name          : Test
    Version       : 11.0.19846.0
    Publisher     : Microsoft
    ExtensionType : ModernDev
    Id            : 8874ed3a-0643-4247-9ced-7a7002f7135d
    Name          : System
    Version       : 11.0.12925.0
    Publisher     : Microsoft
    ExtensionType : ModernDev

Yes we use automatic symbol generation and the server has EnableSymbolLoadingAtServerStartup  enabled.

**Comment by dennisfrostlander** [comment]
Could you please try to re-generate all symbols in C/SIDE by running:
```finsql.exe Command=generatesymbolreference, Database="your-db", ServerName=your-server```
Does the problem still occur after that?

**Comment by sgiordy** [comment]
I have re-generated the symbols many many times :) and... yes, the issue is still present (also with CU2)
You can simply reproduce the issue with the code above.

**Comment by HelleChr** [comment]
I can't reproduce the problem with your code.
Have you tried to Enable Code Analysis for AL to get the exact “problem”.
Se how to do that here: 
https://community.dynamics.com/nav/b/navericwauters/archive/2018/01/22/enable-code-analysis-for-al

**Comment by sgiordy** [comment]
@HelleChr the issue is during the Publish-NAVApp not while creating the package. There is no warning by VSCode. I think is a problem with non-WW localizations.

![al1](https://user-images.githubusercontent.com/17341714/36163082-92d348fa-10e8-11e8-9391-069b0de1e2f4.PNG)
![al2](https://user-images.githubusercontent.com/17341714/36163083-92f21d3e-10e8-11e8-8884-61707e7a3a73.PNG)

**Comment by dennisfrostlander** [comment]
I think I understand what is wrong.
The User table is defined the System symbols package. This package must be published to the server in order for the server-compiler to know what a User table (and all User fields) is.
Get-NavAppInfo showed that you only have Test symbols published on the server, but not System.

You can read more about symbols [here](https://blogs.msdn.microsoft.com/nav/2018/02/02/import-symbols-in-new-or-upgrade-databases-with-microsoft-dynamics-nav-2018/)

**Comment by sgiordy** [comment]
@dennisfrostlander no, System symbols are already loaded. See my previous post.

    Id            : 8874ed3a-0643-4247-9ced-7a7002f7135d
    Name          : System
    Version       : 11.0.12925.0
    Publisher     : Microsoft
    ExtensionType : ModernDev

**Comment by dennisfrostlander** [comment]
Of course, disregard my silly comment before :) I somehow read only the first part of the output.

**Comment by sgiordy** [comment]
Thank you @StanislawStempin I will request for a fix-backport into NAV2018 through Partner Source.

**Comment by StanislawStempin** [comment]
I actually tested it on NAV 2018 CU4 so the issue must already be fixed. Please check that your environment is correctly set up.

**Comment by bleisibach** [comment]
@sgiordy 
I can confirm that with the Cronus DB (CH) from CU 4 the issue is solved.

BUT: When you upgrade an existing DB, in our case from CU2 to CU4, the problem still persits. 

What we detected:
This is how the Cronus DB from DVD (CU4) looks like
![image](https://user-images.githubusercontent.com/36630655/39039941-b904d378-4486-11e8-9a40-27afdf590572.png)

This is how our DB looked like
![image](https://user-images.githubusercontent.com/36630655/39040036-e009ea08-4486-11e8-9ebc-d649854e9604.png)

Symbols for Application are "missing", even after running generatewithsymbols for all objects.

What we did:
***** Note: This is a "hack" - always save your database before you continue. NEVER use in production environement *****

1. Uninstall Test and System App from your DB (Unpublish-NAVapp -ServerInstance XYZ -Name Test (System)
2. Restore the Cronus DB from the DVD (Source)
3. with TSQL, Copy the 3 records (Test, System, Application) from Table [NAV App] and [NAV App Resource] from Cronus DB CU to your destination DB. Take care of the WHERE Clause, do not copy other Apps.

At least, this did the trick for me. I did not face any negative consequences untill now.
